### ISSUE TITLE
If the server fails sending an email the error is not wired back to the client

### ISSUE THREAD

**Issue by GerberI** [SiLeBAT/mibi-portal-client#341]
As an user i would like to be informed if sending a sample sheet fails!!!

Text for toaster: "Es wurde keine E-Mail ans BfR gesendet. Bitte versuchen Sie es später noch einmal."

As a user I want to be informed if my data is not sent to BfR, e.g. because the email address has changed and the portal contains an outdated email address.

Ideas
Email error messages need a little time. Until it arrives, the user might have logged out already. 
- Send email to user which has initiated the MiBi-Portal email by uploading and sending data to a specific BfR lab.
- Send email to admin so that he/she is informed and can change the email address in the system.
- Keep user data in MiBi-Portal and resend to BfR when email has been changed.
- Set banner / toaster "Bitte auf Bestätigung der erfolgreichen Versendung warten".
- Set banner / toaster "Senden erfolgreich" or "Senden fehlgeschlagen".

DevNote: Mail sub-system, Notification Service and Controllers probably need adjusting

**Comment by dptolle** [comment]
@mahtabiltarabian @afalenski  : So far the code changes you did look good, however I noticed a few adjustments we need to discuss.  In the code the sending to the NRLS happens first, so if that fails the system doesn't even try to send to the customer and we get an error message informing the customer that they should try again later (we may want to give them the option to contact support as well). Alex, what should we do if the system manages to send the data to the NRLs, but fails to send the data back to the customer? The error message to the customer should be different - they should be informed that the NRLs go the data but if they require their own copy they should get back to us.  Currently this failure is logged, but we, the admins, never find out about it.  I am not sure how best to handle this ( if email is the problem this would not be the right channel).

**Comment by afalenski** [comment]
@dptolle @mahtabiltarabian Good point! And when emails fails, the option to contact support should be a phone number, shouldn't it?
Suggenstion for the message to the user when NRLs got the email but the user did not: "Ihre Daten wurden ans BfR gesandt. Leider konnte das System keine E-Mail an Sie versenden. Bitte drucken Sie ausnahmsweise die hochgeladene Excel-Datei als Probenbegleitschein aus."

**Comment by mahtabiltarabian** [comment]
@dptolle @afalenski Thanks for the suggestions. My thoughts on each:

Two different messages — agreed, this is the right call. Alex's wording covers the second case well and I'll use it as it is.

Sending to the NRLs first — I would keep it as it is. The customer's copy is the Probenbegleitschein they print and attach to their isolates. If the NRLs never got the order, sending the customer that document would be misleading.

Informing admins by email — I don't think this works. As Dominic says, if email is the problem then email is the one channel we can't use. My suggestion instead: store a flag on the order in the database when the customer copy fails. It works even when mail is down, you can see it in a dashboard at any time, and it gives us the basis to resend the copy later once an address is corrected. I can implement that.

Phone number for support — good idea, I agree. I would put it in the configuration rather than in the code, so it can be changed without a release and differ between QA and production. I just need the number, and whether it should appear in both messages or only when nothing reached the BfR.

**Comment by afalenski** [comment]
Toaster message for the case that no email can be sent: `Das Versenden von E-Mails ist zur Zeit gestört. Bitte rufen Sie den für das MiBi-Portal verantwortliche Person an: [user admin's phone number]`

- Save the user admin's phone number in the config.

**Comment by afalenski** [comment]
@mahtabiltarabian I cannot find the admin phone section in the parse dashboard config section.

**Comment by mahtabiltarabian** [comment]
@afalenski 
Dear Alex, 
Open the Parse Dashboard for the environment (QA )
Left sidebar → Core → Config
Click Create a parameter (below the existing list; a + in some versions)
Fill in:
Name: supportPhone — case-sensitive, must match exactly
Type: String
Value: the real number, up to 30 characters
Save, then reload the page 

<img width="2502" height="219" alt="Image" src="https://github.com/user-attachments/assets/9da2f892-79cf-48de-8d86-015636590a18" />
